lsif-java index reports an error but succeeds with exit code 0

Discovered while auto-indexing the sbt package repo

[sbt.zip](https://github.com/sourcegraph/lsif-java/files/7077604/sbt.zip)

```
unzip sbt.zip
lsif-java index
...
java.lang.RuntimeException: No SemanticDB files found. This typically means that `lsif-java` is unable to automatically index this codebase. If you are using Gradle or Maven, please report an issue to https://github.com/sourcegraph/lsif-java and include steps to reproduce. If you are using a different build tool, make sure that you have followed all of the steps from https://sourcegraph.github.io/lsif-java/docs/manual-configuration.html
```

Same error in other repo maven/org.jboss.xnio/xnio-api@3.8.4.Final
```
[94minfo[39m: Compiling 221 Java sources
Downloading https://github.com/shyiko/jabba/raw/master/index.json
Downloaded https://github.com/shyiko/jabba/raw/master/index.json
Downloading https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/download/jdk8u292-b10/OpenJDK8U-jdk_x64_linux_hotspot_8u292b10.tar.gz
Still downloading:
https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/download/jdk8u292-b10/OpenJDK8U-jdk_x64_linux_hotspot_8u292b10.tar.gz (67.12 %, 69148083 / 103026380)
Downloaded https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/download/jdk8u292-b10/OpenJDK8U-jdk_x64_linux_hotspot_8u292b10.tar.gz
Extracting
  /root/.cache/coursier/v1/https/github.com/AdoptOpenJDK/openjdk8-binaries/releases/download/jdk8u292-b10/OpenJDK8U-jdk_x64_linux_hotspot_8u292b10.tar.gz
in
  /root/.cache/coursier/jvm/adopt@1.8.0-292
Done
[94minfo[39m: $ /root/.cache/coursier/jvm/adopt@1.8.0-292/bin/javac @/data/target/javacopts.txt
[94minfo[39m: /tmp/lsif-java2085231075809640276/s/org/xnio/_private/Messages_$logger.java:54: error: duplicate class: org.xnio._private.Messages_$logger
[94minfo[39m: public class Messages_$logger extends DelegatingBasicLogger implements Messages, BasicLogger, Serializable {
[94minfo[39m:        ^
[94minfo[39m: /data/org/xnio/Xnio.java:43: error: package org.osgi.framework does not exist
[94minfo[39m: import org.osgi.framework.Bundle;
[94minfo[39m:                          ^
[94minfo[39m: /data/org/xnio/Xnio.java:44: error: package org.osgi.framework does not exist
[94minfo[39m: import org.osgi.framework.BundleContext;
[94minfo[39m:                          ^
[94minfo[39m: /data/org/xnio/Xnio.java:45: error: package org.osgi.framework does not exist
[94minfo[39m: import org.osgi.framework.FrameworkUtil;
[94minfo[39m:                          ^
[94minfo[39m: /data/org/xnio/Xnio.java:46: error: package org.osgi.framework does not exist
[94minfo[39m: import org.osgi.framework.ServiceReference;
[94minfo[39m:                          ^
[94minfo[39m: 5 errors
```

--- a/lsif-java/src/main/scala/com/sourcegraph/lsif_java/buildtools/LsifBuildTool.scala
+++ b/lsif-java/src/main/scala/com/sourcegraph/lsif_java/buildtools/LsifBuildTool.scala
@@ -165,11 +165,11 @@ class LsifBuildTool(index: IndexCommand) extends BuildTool("LSIF", index) {
       return CommandResult(0, Nil)
     }
 
-    val compileAttemtps = ListBuffer.empty[Try[Unit]]
-    compileAttemtps += compileJavaFiles(tmp, deps, config, javaFiles)
-    compileAttemtps += compileScalaFiles(deps, scalaFiles)
-    compileAttemtps += compileKotlinFiles(deps, config, kotlinFiles)
-    val errors = compileAttemtps.collect { case Failure(exception) =>
+    val compileAttempts = ListBuffer.empty[Try[Unit]]
+    compileAttempts += compileJavaFiles(tmp, deps, config, javaFiles)
+    compileAttempts += compileScalaFiles(deps, scalaFiles)
+    compileAttempts += compileKotlinFiles(deps, config, kotlinFiles)
+    val errors = compileAttempts.collect { case Failure(exception) =>
       exception
     }
 
@@ -179,6 +179,9 @@ class LsifBuildTool(index: IndexCommand) extends BuildTool("LSIF", index) {
     val isSemanticdbGenerated = Files
       .isDirectory(targetroot.resolve("META-INF"))
     if (errors.nonEmpty && !isSemanticdbGenerated) {
+      errors.foreach { error =>
+        index.app.reporter.log(Diagnostic.exception(error))
+      }
       CommandResult(1, Nil)
     } else {
       if (errors.nonEmpty && isSemanticdbGenerated) {
@@ -376,8 +379,9 @@ class LsifBuildTool(index: IndexCommand) extends BuildTool("LSIF", index) {
   )(fn: PresentationCompiler => T): T = {
     val scalaVersion = deps
       .classpath
-      .headOption
+      .iterator
       .flatMap(jar => ScalaVersion.inferFromJar(jar))
+      .nextOption()
       .getOrElse {
         throw new IllegalArgumentException(
           s"failed to infer the Scala version from the dependencies: " +

